sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +703,65 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+//TODO: Timed counterpart for timed AssetLockListener for reversion
+const assetLockCallbackTimed = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    //Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+
+    const reclaimCallback = async function(contract: Contract, assetType: string, assetID: string, recipientECert: string, contractID: string) {
+        //TODO: Call isAssetLockedInHTLC -> Hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract,contractID));
+        
+        if(islocked == false) {
+            //CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            //CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                //throw new Error(`GetHTLCHashPreImageByContractId evaluateTransaction Error: ${evaluateError}`);
+                //If it succeeds, no change
+                const [reclaimResult, isReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                if(isReclaimableQueryError) {
+                    let i = 0;
+                    do {
+                        reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);
```

Comment:
Here also take the return and check if there is no error then break out of loop. No need to reclaim again if 1st or 2nd retry was succesfull

Reply:
@sanvenDev asked in chat how to do this given promise<any> return type disparity

Reply:
use the same logic what you used in the 1st attempt for reclaim before the loop (using handlePromise that returns error) . There you can have check if there's error then continue else break